### DISCUSSION TITLE
RFE: vectors in different languages should be aligned

### DISCUSSION THREAD

**Discussion by asterbini** [explosion/spaCy#5888] [New Features & Project Ideas]
## Feature description
When working on different languages it would be very useful to compare word vectors to compare their "meaning".
Yet different models seems to use vector spaces that are not aligned, thus vectors for words with the same meaning are normally not similar.

It would be nice if:
- all models released by Spacy are aligned, and thus similar-meaning lemmas have similar vectors.
- or else all models have a transformation matrix that, applied to the current space, does transform it to a common vector space (e.g. 'en')

**Comment by honnibal** [comment]
Agree that this would be nice but it would mean training the various models jointly, and it may impact the accuracy of specific models (especially models for smaller languages). It's also a big task to set up evaluation of the alignment.

If someone wants to look at this as a research project and they can get convincing results, we'd consider adopting it. But I think it seems like quite an ambitious project that won't yield results that are that compelling in practice.

Instead if your specific project requires this, you'll want to train your own vector models.

**Comment by asterbini** [comment]
Right, I'll look for alignment algorithms or for how to produce the transformation matrices to align two models.
(https://arxiv.org/abs/1804.07745)

**Comment by asterbini** [comment]
I have found the MUSE project from Facbook Research https://github.com/facebookresearch/MUSE
